src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -242,4 +303,51 @@ Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         return value;
     }
   }
+  public enum BehaviorOnNullValues {
+    IGNORE,
+    DELETE,
+    FAIL;
+
+    public static final BehaviorOnNullValues DEFAULT = IGNORE;
```

Comment:
If we're aiming for backwards compatibility ("The FAIL option is identical to the previous behavior, so this option is the default to ensure backward compatibility for existing connector deployments that are upgrading") this should be changed to `DEFAULT = FAIL;`.

Reply:
I think it's safe to enable a sane config as the default (here `IGNORE`). Backwards compatibility, as I see it here, means that if people have built software around the assumption that the connector fails, they can still enable this behavior. 

But we should have the connector, starting with this version, not failing when it encounters records with non-null key/null value.

Reply:
I'm fine with changing the documentation, since the previous behavior is not really a useful default. @C0urante, do you want to make those changes and push them to this branch?

Reply:
Which documentation changes are you referring to? `configuration_options.rst` already lists `ignore` as the default, not sure where else would need to be changed that hasn't already been.

Reply:
Sorry, that comment was in the commit. Let me amend the commit message.